soccer/planning/RRTPlanner.cpp

```diff
@@ -182,57 +182,28 @@ vector<Point> RRTPlanner::runRRT(MotionInstant start, MotionInstant goal,
                                  const ShapeSet& obstacles) {
     // unique_ptr<InterpolatedPath> path = make_unique<InterpolatedPath>();
 
-    // Initialize two RRT trees
-    FixedStepTree startTree;
-    FixedStepTree goalTree;
-    startTree.init(start.pos, &obstacles);
-    goalTree.init(goal.pos, &obstacles);
-    startTree.step = goalTree.step = .15f;
-
-    // Run bi-directional RRT algorithm
-    Tree* ta = &startTree;
-    Tree* tb = &goalTree;
-    for (unsigned int i = 0; i < _maxIterations; ++i) {
-        Geometry2d::Point r = RandomFieldLocation();
-
-        Tree::Point* newPoint = ta->extend(r);
-
-        if (newPoint) {
-            // try to connect the other tree to this point
-            if (tb->connect(newPoint->pos)) {
-                // trees connected
-                // done with global path finding
-                // the path is from start to goal
-                // runRRT will handle the rest
-                break;
-            }
-        }
+    // Initialize bi-directional RRT
+    auto stateSpace = make_shared<RoboCupStateSpace>(
```

Comment:
Alternatively, you can save the stateSpace instead of making a new shared_pointer each iteration.